src/accelerate/accelerator.py

```diff
@@ -1456,15 +1461,17 @@ def prepare_model(self, model: torch.nn.Module, device_placement: bool = None, e
                             ),
                             auto_wrap_policy=fsdp_plugin.auto_wrap_policy,
                         )
-
+                # if the previous and current models are same, delete the previous one
+                if len(self._models) > 1 and (self._models[-2] is self._models[-1]):
+                    del self._models[-2]
```

Comment:
Instead of fixing this here, would it be possible to prevent adding the identical model twice?

Reply:
Currently, I don't see a way of avoiding it, will circle back to this in a few days